src/kernel/arch/x86_64/arch.zig

```diff
@@ -0,0 +1,458 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const log = std.log.scoped(.x86_64_arch);
+const serial = @import("serial.zig");
+const stivale2 = @import("stivale2.zig");
+const tty = @import("tty.zig");
+const mem = @import("../../mem.zig");
+const vmm = @import("../../vmm.zig");
+const Serial = @import("../../serial.zig").Serial;
+const panic = @import("../../panic.zig").panic;
+const MemProfile = mem.MemProfile;
+
+/// The memory type provided by the Limine bootloader.
+const MemType = enum(u32) {
+    Usable = 1,
+    Reserved = 2,
+    ACPI_Reclaimable = 3,
+    ACPI_NVS = 4,
+    BadMemory = 5,
+    BootloaderReclaimable = 0x1000,
+    KernelAndModules = 0x1001,
+};
+
+/// The interrupt context that is given to a interrupt handler. It contains most of the registers
+/// and the interrupt number and error code (if there is one).
+pub const CpuState = packed struct {
+    // Page directory
+    cr3: usize,
+    // Extra segments
+    gs: u32,
+    fs: u32,
+    es: u32,
+    ds: u32,
+
+    // Destination, source, base pointer
+    edi: u32,
+    esi: u32,
+    ebp: u32,
+    esp: u32,
+
+    // General registers
+    ebx: u32,
+    edx: u32,
+    ecx: u32,
+    eax: u32,
+
+    // Interrupt number and error code
+    int_num: u32,
+    error_code: u32,
+
+    // Instruction pointer, code segment and flags
+    eip: u32,
+    cs: u32,
+    eflags: u32,
+    user_esp: u32,
+    user_ss: u32,
```

Comment:
Shouldn't most of these be u64 (usize)?

Reply:
Have removed this as will be different for 64 bit